frontend/src/Components/CooperativeGame.tsx

```diff
@@ -0,0 +1,140 @@
+import { useEffect, useState } from 'react'
+import '../Styles/SoloGame.css'
+import { useParams } from "react-router-dom";
+import incorrectButton from '../Sounds/error-8-206492.mp3'
+import correctButton from '../Sounds/new-notification-7-210334.mp3'
+import { onSnapshot, doc, updateDoc } from 'firebase/firestore';
+import { db } from '../FireBase/firebase-config';
+import Cookies from "universal-cookie";
+
+const cookies = new Cookies();
+
+function CooperativeGame() {
+
+  const coresDisponiveis = ['Vermelho', 'Amarelo', 'Verde', 'Azul'];
+  const [cores, setCores] = useState<string[]>([])
+  const [ultimaCor, setUltimaCor] = useState('')
+  const [escolhasDoPlayer, setEscolhasDoPlayer] = useState<string[]>([]);
+  var [rodada, setRodada] = useState(1)
+
+  var numeroAleatorio = Math.floor(4 * Math.random())
+  const corSelecionada = coresDisponiveis[numeroAleatorio];
+//--------------------------------------------------------------------------------------
+  const [playersInfos, setPlayersInfos] = useState({player1Img:'', player2Img:'',player1Name:'', player2Name:''});
+
+
+  const [gameInfo, setGameInfo] = useState({currentPlayer: ''});
+
+  const { roomname }:any = useParams();
+
+  function TocarAudio(audio:string) {
```

Comment:
This one should be put in another file bcz of the principle of single responsability.